Use the latest version of the AWS provider for Terraform

jsf9k opened the pull request:
## 🗣 Description ##

This pull request upgrades to using the latest version of the Terraform AWS provider.  It also commits the `.terraform.lock.hcl` file for more reproducible builds.

## 💭 Motivation and context ##

We should use a modern Terraform AWS provider wherever we can; otherwise, we will get left behind.

Adding the `.terraform.lock.hcl` file is a recommended practice, and it will lead to more reproducible builds.  See [the Terraform documentation](https://developer.hashicorp.com/terraform/language/files/dependency-lock) for more details.

## 🧪 Testing ##

All automated tests pass.

I verified that this version works for this repository by re-running `terraform apply` for our `dev-a` deployment for all the subdirectories in this repository.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.
- [x] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release (necessary if and only if the version was bumped).

jsf9k committed:
Use the latest AWS provider for Terraform

I verified that this version works for this repository by re-running
terraform apply for our dev-a deployment for all the subdirectories in
this repository.

jsf9k committed:
Update README.md files for Terraform AWS provider changes

jsf9k commented:
@mcdonnnj - It's unclear to me if this PR warrants a version bump or not.

mcdonnnj reviewed (approved):
LGTM ✔ Since we are updating a dependency file I would say that it would be fine to get a version bump.

jsf9k committed:
Bump version from 1.0.0 to 1.0.1

jsf9k committed:
Remove .terraform.lock.hcl from .gitignore

jsf9k committed:
Add .terraform.lock.hcl files

Using this file results in more reproducible builds and protects us
against the insertion of a malicious provider.  See
https://developer.hashicorp.com/terraform/language/files/dependency-lock
for more details.

dav3r reviewed (approved):
👍 👍 👍

mcdonnnj reviewed (approved):
Still lookin' good to me ✔ One thing is that the `.gitignore` change should be made in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module).

jsf9k commented:
> Still lookin' good to me ✔ One thing is that the `.gitignore` change should be made in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module).

Yes, I thought about that.  I will continue down the path I'm on _and_ create a PR in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module) as well.

jsf9k commented:
> > Still lookin' good to me ✔ One thing is that the `.gitignore` change should be made in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module).
> 
> Yes, I thought about that. I will continue down the path I'm on _and_ create a PR in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module) as well.

Please see cisagov/skeleton-tf-module#247.
